mlir_graphblas/src/lib/GraphBLAS/GraphBLASLowerPass.cpp

```diff
@@ -1669,6 +1683,191 @@ class LowerMatrixMultiplyRewrite
   };
 };
 
+class MatrixMultiplyGenericDWIMFirstArgRewrite
+    : public OpRewritePattern<graphblas::MatrixMultiplyGenericOp> {
+public:
+  using OpRewritePattern<graphblas::MatrixMultiplyGenericOp>::OpRewritePattern;
+
+  template <class T>
+  static bool needsDWIM(T op) {
+    Value A = op.a();
+    RankedTensorType aType = A.getType().cast<RankedTensorType>();
+    int64_t aRank = aType.getRank();
+    if (aRank == 2) {
+      bool aIsCSC = hasColumnOrdering(aType);
+      if (aIsCSC) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  LogicalResult match(graphblas::MatrixMultiplyGenericOp op) const override {
```

Comment:
Is there a strong reason to separate `match` from `rewrite`?

Reply:
For complex matching, I agree with the separation of concerns. But for this level of checking, which amounts to a single `if` statement, I would rather keep is more concise.